src/pages/bubblegum/guides/javascript/how-to-create-1000000-nfts-on-solana.md

```diff
@@ -0,0 +1,695 @@
+
+import fs from 'fs'
+import { irysUploader } from '@metaplex-foundation/umi-uploader-irys'
+
+// Create the wrapper function
+const createCnft = async () => {
+  ///
+  ///
+  ///  all our code will go in here
+  ///
+  ///
+}
+
+// run the wrapper function
+createCnft()
+```
+
+## Setting up Umi
+
+This example is going to run through setting up Umi with a `generatedSigner()`. If you wish to try this example with React you'll need to setup Umi via the `React - Umi w/ Wallet Adapter` guide. Apart from the the wallet setup this guide will for fileStorage keys and wallet adapter.
+
+### Generating a New Wallet
+
+```ts
+const umi = createUmi('https://api.devnet.solana.com')
+  .use(mplBubblegum())
+  .use(mplTokenMetadata())
+  .use(
+    irysUploader({
+      // mainnet address: "https://node1.irys.xyz"
+      // devnet address: "https://devnet.irys.xyz"
+      address: 'https://devnet.irys.xyz',
+    })
+  )
+
+const signer = generateSigner(umi)
+
+umi.use(signerIdentity(signer))
+
+// This will airdrop SOL on devnet only for testing.
+console.log('Airdropping 1 SOL to identity')
+await umi.rpc.airdrop(umi.identity.publickey, sol(5))
+```
+
+### Use an Existing Wallet Locally
+
+```ts
+const umi = createUmi('https://api.devnet.solana.com')
+  .use(mplBubblegum())
+  .use(mplTokenMetadata())
+  .use(
+    irysUploader({
+      // mainnet address: "https://node1.irys.xyz"
+      // devnet address: "https://devnet.irys.xyz"
+      address: 'https://devnet.irys.xyz',
+    })
+  )
+
+// Generate a new keypair signer.
+const signer = generateSigner(umi)
+
+// You will need to us fs and navigate the filesystem to
+// load the wallet you wish to use via relative pathing.
+const walletFile = fs.readFileSync('./keypair.json')
+
+// Convert your walletFile onto a keypair.
+let keypair = umi.eddsa.createKeypairFromSecretKey(new Uint8Array(walletFile))
+
+// Load the keypair into umi.
+umi.use(keypairIdentity(keypair))
+```
+
+## Creating an cNFT
+
+Creating a cNFT on Solana is fairly simple and requires a few items to get ready before we can actually perform the minting and reading operations.
+
+- A Merkle tree to store our cNFT data to.
+- A DAS ready RPC to be able to read the data from an indexer that is storing our data during creation.
+
+#### Merkle Tree
+
+A Merkle Tree for the most park can be thought of as a "database" of cNFT data. A Merkle Tree is created and cNFTs can be added to it until it's full.
+
+#### DAS RPCs
+
+Due to the nature of a Merkle Tree cNFT data isn't stored in Solana accounts and instead is stored in the ledger state. To be able to read the data back effectively we need to use an indexer which indexes all the cNFT data as its created/mutated. DAS enabled RPCs are RPCs that are running the DAS indexer service and allow us to query the RPC provider for this data on demand.
+
+DAS RPC providers include:
+
+- Helius
+- Extrnode
+- Quicknode
+- Shyft
+- Hellomoon
+- Triton
```

Comment:
I'd prefer to link the rpc page here so that we dont have to update the list on multiple places?